src/internal/components/pluggableVisualizations/bulletChart/bucketHelper.ts

```diff
@@ -0,0 +1,60 @@
+// (C) 2020 GoodData Corporation
+import * as BucketNames from "../../../../constants/bucketNames";
+import { IBucket, IBucketItem } from "../../../interfaces/Visualization";
+import { METRIC } from "../../../constants/bucket";
+import {
+    getPreferredBucketItems,
+    getAllAttributeItems,
+    limitNumberOfMeasuresInBuckets,
+} from "../../../utils/bucketHelper";
+
+const orderedMeasureBucketsLocalIdentifiers = [
+    BucketNames.MEASURES,
+    BucketNames.SECONDARY_MEASURES,
+    BucketNames.TERTIARY_MEASURES,
+];
+
+const placeMeasures = (buckets: IBucket[]) => {
+    const emptyBucketsLocalIdentifiers: string[] = [];
+    const bucketsWithPlacedMeasure: IBucket[] = [];
+    const measuresToBePlaced: IBucketItem[] = [];
+
+    orderedMeasureBucketsLocalIdentifiers.forEach((bucketLocalIdentifier: string) => {
+        const preferredBucketItems = getPreferredBucketItems(buckets, [bucketLocalIdentifier], [METRIC]);
+
+        if (preferredBucketItems.length === 0) {
+            emptyBucketsLocalIdentifiers.push(bucketLocalIdentifier);
```

Comment:
could you just add this bucket with empty items a then run through all buckets and fill those that are empty?
Then you don't need to sort it at the end